packages/synapse-react-client/src/components/CreateTableViewWizard/CreateTableViewWizard.tsx

```diff
@@ -0,0 +1,359 @@
+ */
+export default function CreateTableViewWizard(
+  props: CreateTableViewWizardProps,
+) {
+  const { open, parentId, onComplete, onCancel } = props
+  const [step, setStep] =
+    useState<CreateTableViewWizardStep>('CHOOSE_TABLE_TYPE')
+
+  const [name, setName] = useState('')
+  const [description, setDescription] = useState<string | undefined>(undefined)
+
+  const [entityType, setEntityType] = useState<EntityType | undefined>(
+    undefined,
+  )
+
+  const [entityViewScopeIds, setEntityViewScopeIds] = useState<string[]>([])
+  const [submissionViewScopeIds, setSubmissionViewScopeIds] = useState<
+    string[]
+  >([])
+  const [viewTypeMask, setViewTypeMask] = useState<number>(
+    ENTITY_VIEW_TYPE_MASK_FILE,
+  )
+  const isProjectView = viewTypeMask === ENTITY_VIEW_TYPE_MASK_PROJECT
+
+  const [columnModels, setColumnModels] = useState<
+    SetOptional<ColumnModel, 'id'>[]
+  >([])
+  const columnSchemaFormRef = useRef<SubmitHandle>(null)
+
+  const [sql, setSql] = useState('')
+
+  const { mutate: createEntity, error: createEntityError } = useCreateEntity({
+    onSuccess: entity => {
+      onComplete(entity.id!)
+    },
+  })
+
+  const { mutateAsync: createColumnModels, error: createColumnModelsError } =
+    useCreateColumnModels()
+
+  const onTableTypeSelected = useCallback(
+    (entityType: EntityType, viewTypeMask?: number) => {
+      setEntityType(entityType)
+      if (viewTypeMask != null) {
+        setViewTypeMask(viewTypeMask)
+      }
+      setStep(getStepAfterTypeSelection(entityType))
+    },
+    [],
+  )
+
+  const onViewSelected = useCallback(() => {
+    setStep('CHOOSE_VIEW_TYPE')
+    // Reset the scope IDs in case we switched from an EntityView to a SubmissionView
+  }, [])
+
+  const onFinish = useCallback(async () => {
+    if (!entityType) {
+      console.error('entityType was not defined!')
+      setStep('CHOOSE_TABLE_TYPE')
+      return
+    }
+    // Create the column models
+    let createdColumnModels: ColumnModel[] = []
+    if (
+      columnModels &&
+      // Do not create columns if this is a SQL-defined view
+      entityType !== EntityType.MATERIALIZED_VIEW &&
+      entityType !== EntityType.VIRTUAL_TABLE
+    ) {
+      try {
+        createdColumnModels = await createColumnModels(columnModels)
+      } catch (e) {
+        console.error('Error creating column models', e)
+        return
+      }
+    }
+    // Create the entity
+    const entityToCreate: Entity = {
+      name,
+      description,
+      parentId: parentId,
+      concreteType: convertToConcreteEntityType(entityType),
+    }
+
+    maybeSetColumnIds(entityToCreate, entityType, createdColumnModels)
+    maybeSetScopeIds(
+      entityToCreate,
+      entityType,
+      entityType === EntityType.SUBMISSION_VIEW
+        ? submissionViewScopeIds
+        : entityViewScopeIds,
+    )
+    maybeSetViewTypeMask(entityToCreate, entityType, viewTypeMask)
+    maybeSetDefiningSQL(entityToCreate, entityType, sql)
```

Comment:
Conditionally add each field based on entity type--but delegate the logic to the utils for readability